### DISCUSSION TITLE
How to generate a case when in a view

### DISCUSSION THREAD

**Discussion by Foxsterc** [simolus3/drift#2626] [Q&A] [answered]
Hi

How would you write this a a view in drift?

`    SELECT t0."id",
         case when  t0."date_attempted" is null then 1 else 0 end as Completed
      FROM logbooks "t0"
     GROUP BY t0."id"
     `

I've tried 

`  Query as() => select([
        logbooks.Id,
        logbooks.dateAttempted.caseMatch<bool>(
          when: {
            const Constant(null): const Constant(true),
          },
          orElse: const Constant(false),
        )
      ]).from(logbooks ).join([])
        ..groupBy([
          logbooks.Id,
        ]);`

But this doesn't generate a case when in the db.

Thanks,

**Comment by simolus3** [comment, marked as answer]
View columns need to have a name so that they can be referred to by other selects reading from the view. You can add custom columns by writing getters in the view class:

```dart
Expression<bool> get hasAttempt = logbooks.dateAttempted.caseMatch(...);

@override
Query as() => select([logbooks.id, hasAttempt]).from(...);
```

**Reply by Foxsterc** [reply]
That's spot on thank you!
